### ISSUE TITLE
Unknown layer: SpatialDropout2D

### ISSUE THREAD

**Issue by josemafuen** [tensorflow/tfjs#1453]
The following error appears to me when trying to import a model in tensorflow.js

Uncaught (in promise) Error: Unknown layer: SpatialDropout2D. This may be due to one of the following reasons:
1. The layer is defined in Python, in which case it needs to be ported to TensorFlow.js or your JavaScript code.
2. The custom layer is defined in JavaScript, but is not registered properly with tf.serialization.registerClass().

It seems like SpatialDropout isn't implemented yet. Is anyone working on it?

**Comment by syt123450** [comment]
Hi @caisq , are you working on it? I would love to implement SpatialDropout2D layer and noise shape.

**Comment by jessetrana** [comment]
In case it helps anyone with preexisting models: I found this issue because I had trained a model in Keras, used tensorflowjs_converter, and then encountered this error only when trying to infer in JS. In this situation, you can remove the layer in the converted model by simply modifying model.json to remove the layer reference and pull the output of the previous layer to be the input of the next. That way you don't have to do the surgery in Keras as a post-processing step on model saving.

**Comment by syt123450** [comment]
Hi @caisq , I took a look into this issue, it seems that to implement `SpatialDropout2D`, we need to make `tf.layers.dropout` support `noiseShape` first. While dropout layer only supports default `noiseShape` now (as `dropout` op in `tfjs-core` only support default `noiseShape`), I sent two PRs to make the `dropout` op and `dropout` layer support non-default `noiseShape`. Could you please take a look into these PRs? Thanks!

* Make dropout op support non-default noiseShape [tensorflow/tfjs-core#1782](https://github.com/tensorflow/tfjs-core/pull/1782)
* Make dropout layer support non-default noiseShape and seed [tensorflow/tfjs-layers#556](https://github.com/tensorflow/tfjs-layers/pull/556)
